src/components/CodeEditorHeader.tsx

```diff
@@ -40,6 +40,13 @@ export const CodeEditorHeader = ({
   const [footprintDialogOpen, setFootprintDialogOpen] = useState(false)
   const { toast } = useToast()
 
+  // Check if manual-edits.json has any content
+  const [hasManualEdits, setHasManualEdits] = useState(false)
+
+  useEffect(() => {
+    setHasManualEdits(files["manual-edits.json"].length > 0)
+  }, [files["manual-edits.json"]])
```

Comment:
this is a DOUBLE anti-pattern 😁 

1. Never use `useEffect` and `useState` when there's nothing async, just use `useMemo`
2. Just use variables if there's no cost to memoization!!!

```ts
const hasManualEdits = files["manual-edits.json"].length > 0
```